### ISSUE TITLE
[unused_private_declaration] False positives on CodingKeys

### ISSUE THREAD

**Issue by kimdv** [realm/SwiftLint#2573]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.30.0
* Installation method used: Homebrew
* Paste your configuration file:

```yml
disabled_rules: # rule identifiers to exclude from running
  - trailing_whitespace
  - comma
  - line_length
  - todo
  - identifier_name

opt_in_rules: # some rules are only opt-in
#  - empty_count
  # Find all the available rules by running:
  # swiftlint rules
included: # paths to include during linting. `--path` is ignored if present.
  - AppName

excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Carthage

analyzer_rules:
  - unused_private_declaration
  - unused_import

type_body_length:
  - 300 # warning
  - 500 # error

file_length:
  warning: 500
  error: 1200

type_name:
  max_length: # warning and error
    warning: 50
    error: 60

identifier_name:
  max_length: # warning and error
    warning: 60
    error: 70
  excluded: # excluded via string array
    - id
    - db

```

```swift
// This triggers a violation:
struct ResponseModel: Codable {
    let items: [Item]

    private enum CodingKeys: String, CodingKey { // <-- False 'unused_private_declaration' 
        case items = "ResponseItems"
    }
}
```

**Comment by kimdv** [comment]
I see your point! 
Where will you have the reporting? 😄 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jpsim** [comment]
> we should stop reporting violations here

Oh I meant the rule should stop reporting violations on `CodingKeys` enum that conform to a `CodingKeys` protocol inside a `Codable` container.

You definitely filed this issue at the right place! 😅

**Comment by kimdv** [comment]
Oh sorry for misunderstanding. haha! 😆
